lib/features/search/email/presentation/search_email_controller.dart

```diff
@@ -282,36 +251,93 @@ class SearchEmailController extends BaseController
         }
       }
     );
+
+    emailUIActionWorker = ever(
+      mailboxDashBoardController.emailUIAction,
+      (action) {
+        if (action is RefreshChangeEmailAction) {
+          _refreshEmailChanges(newState: action.newState);
+        }
+      },
+    );
   }
 
-  void _onSearchTextInputListener() {
-    if (textInputSearchFocus.hasFocus) {
-      searchIsRunning.value = false;
+  void _refreshEmailChanges({jmap.State? newState}) {
+    log('SearchEmailController::_refreshEmailChanges(): newState: $newState');
+    if (accountId == null ||
+        session == null ||
+        mailboxDashBoardController.currentEmailState == null ||
+        newState == null ||
+        searchIsRunning.isFalse) {
+      return;
     }
+
+    _webSocketQueueHandler?.enqueue(WebSocketMessage(newState: newState));
+  }
+
+  void _initWebSocketQueueHandler() {
+    _webSocketQueueHandler = WebSocketQueueHandler(
+      processMessageCallback: _handleWebSocketMessage,
+      onErrorCallback: onError,
```

Comment:
because it is only related to UI, please turn it on whenever user in search? WDYT?

Reply:
Yes, it is working like that. It will go along with the lifecycle of SearchEmailController. SearchEmailController is only instantiated when the user moves into the search view. Exiting the search view it will be disposed.